osmaxx/excerptexport/static/excerptexport/scripts/maps/draw_controls.js

```diff
@@ -0,0 +1,129 @@
+var draw_controls = function (map) {
+    // holds the geoJSON text representation
+    var geoJSON_element = document.getElementById('id_bounding_geometry');
+    geoJSON_element.parentNode.parentNode.className = "hidden";
+
+    var editableLayers = new L.FeatureGroup();
+    map.addLayer(editableLayers);
+
+    var draw_enabled = {
+        polyline: false,
+        circle: false, // Turns off this drawing tool
+        marker: false,
+        polygon: {
+            allowIntersection: false, // Restricts shapes to simple polygons
+            drawError: {
+                color: '#e1e100', // Color the shape will turn when intersects
+                message: '<strong>Oh snap!<strong> you can\'t draw that!' // Message that will show when intersect
+            },
+            shapeOptions: {
+                color: 'black',
+                fillOpacity: 0.15
+            }
+        },
+        rectangle: {
+            shapeOptions: {
+                color: 'black',
+                fillOpacity: 0.15
+            }
+        }
+    };
+    var options = {
+        position: 'topright',
+        draw: draw_enabled,
+        edit: {
+            featureGroup: editableLayers,
+            remove: true
+        }
+    };
+    var optionsDisabled = {
+        position: 'topright',
+        draw: false,
+        edit: {
+            featureGroup: editableLayers,
+            remove: true
+        }
+    };
+
+    var drawControlEnabled = new L.Control.Draw(options);
+    var drawControlDisabled = new L.Control.Draw(optionsDisabled);
+
+    drawControlEnabled.addTo(map);
+    function updateGeoJSON (layer){
+        geoJSON_element.value = JSON.stringify(layer.toGeoJSON()['geometry']);
+    }
+
+    function showSizeOfLayer(layer) {
+        var allowedMaxSize = 5 * 1000 * 1000 * 1000;  // 5 GB in bytes, approx. Germany
+        var nameField = document.getElementById('id_name');
+
+        var e = document.getElementById('error_size_estimation_too_large');
+        if (!e) {
+            e = document.createElement('span');
+            e.className = 'help-block';
+            e.id = 'error_size_estimation_too_large';
+            nameField.parentNode.appendChild(e);
+        }
+
+        estimateSize(layer).done(function (data) {
+            function formatBytes(bytes) {
+               if(bytes == 0) return '0 Byte';
+               var k = 1000;
+               var sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+               var i = Math.max(Math.floor(Math.log(bytes) / Math.log(k)));
```

Comment:
Uh, `max` alone won't future-prove this. [The `, sizes.length - 1` part](https://github.com/geometalab/osmaxx-frontend/pull/472#discussion-diff-62335502) would be quite essential ;-)